core-bundle/src/Repository/CronRepository.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Repository;
+
+use Contao\CoreBundle\Entity\Cron as CronEntity;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @internal
+ */
+class CronRepository extends ServiceEntityRepository
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, CronEntity::class);
+
+        if (($connection = $registry->getConnection()) instanceof Connection) {
+            $this->connection = $connection;
+        }
+    }
+
+    public function lockTable(): void
+    {
+        $table = $this->getClassMetadata()->getTableName();
+
+        $this->connection->exec("LOCK TABLES $table WRITE, $table AS t0 WRITE, $table AS t0_ WRITE");
```

Comment:
Why do you have to lock aliases here?

Reply:
The `RememberMeRepository` does the same, otherwise there will be an error. 

```
An exception occurred while executing 'SELECT t0.id AS id_1, t0.name AS name_2, t0.lastRun AS lastRun_3 FROM tl_cron t0 WHERE t0.name = ? LIMIT 1' with params ["App\\Cron\\ExampleCliCron::__invoke"]:

SQLSTATE[HY000]: General error: 1100 Table 't0' was not locked with LOCK TABLES 
```

Weirdly, I had to use `t0` instead of `t0_` 🤷‍♂ .